### DISCUSSION TITLE
File search worker Eclipse error

### DISCUSSION THREAD

**Discussion by axbusch** [Genuitec/Copilot4Eclipse-community#19] [Help] [answered]
Hi,

i have two different Eclipse based IDEs (Cube MX studio 1.16 and Reneses e2 Studio) where i now cannot search anymore in my project. 
i get a file search worker error

```
!MESSAGE An internal error occurred during: “File Search Worker”.
!STACK 0
java.lang.NoSuchMethodError: 'byte[] org.eclipse.core.resources.IFile.readNBytes(int)'

```
there are error logs and it is the same error i face.
https://community.st.com/t5/stm32cubeide-mcus/search-an-internal-error-occurred-during-quot-file-search-worker/td-p/720854

someone there said it happend after he installed the copilot plugin. i have the plugin installed also in both IDEs but it worked for months without problems. So this is not fingerpointing (just to be clear)

i just wanted to ask if you or somebody else had this bug, too? or if anybody has an idea how to fix this?

Regards

**Comment by wayneparrott** [comment]
Thanks for sharing this info.
The missing method for IFile has been around for a few years. So there is something goofy happening on these systems that I can not directly relate to Copilot4Eclipse atm. I fired up my STM32CubeIDE v1.16.0 w/ Copilot4Eclipse v1.2.1. I checked for updates - none presented. ThenI  ran a few Find (ctl-f) and Search menu > File tests with no issues observed. 

Do you recall if you recently updated your Cube IDE or if it automatically updated recently?

An experiment to run is to install a new clean version of Cube and run search/find tests. If all OK, check for updates and accept. Then run search/find tests again. If all OK install Copilot4Eclipse and retest Find scenarios.

W

**Reply by axbusch** [reply]
in the renesas IDE some AI plugin was updated but this seems unreleated to me. 
in cubeMX i had no updates.
A colleague who had this error also reinstalled the cubeMX IDE and the error was gone but he did not install the copilot plugin since.

but as you said: i found some forum error logs with this File Search Worker from a time where no copilot existed so i would also think the copilot plugin is not the root cause

**Reply by skutter-x86** [reply]
I tried several times with MCUXpresso, installing/uninstalling the IDE and the problem with Search starts only when I install Copilot4Eclipse. In that case it's easy to reproduce.

Uninstalling the plugin does not solve the issue, you must reinstall the IDE. Something gets corrupted inside.

**Comment by wayneparrott** [comment]
Update: I am able to replicate this issue. Researching it now.

**Reply by skutter-x86** [reply]
I have the very same problem with MCUXpresso IDE v11.10.0 and Copilot4Eclipse 1.2.1. After I installed Copilot4Eclipse, not immediately, but shortly after, my File search broke with the same error.

**Reply by axbusch** [reply]
More colleagues who use Stm32CubeIDE and the copilot plugin are now facing this error.
As far as i know without any updates. It did work without problems for weeks (or months) and now one by one they get this bug.
For now the only bugfix i know is reinstall of the IDE and then they are hesitating to install the plugin again :(
atm i use notepad++ for searching my project

**Comment by wayneparrott** [comment, marked as answer]
This issue has been fixed. Please read https://github.com/Genuitec/Copilot4Eclipse-community/discussions/23

**Reply by axbusch** [reply]
Top, this is great!

**Reply by skutter-x86** [reply]
Confirmed to be working flawlessly in MCUXpresso IDE. Thank you very much for such a quick fix!
